### DISCUSSION TITLE
[PorousFlowPiecewiseLinearSink] want to dig deeper into the drain BCs for mining application

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#19651] [Q&A Modules: General] [answered]
Hi guys, so I did some tests on draining the water out of the excavated block (similar to https://mooseframework.inl.gov/modules/porous_flow/coal_mining.html done by @WilkAndy)

Below are the geometries, I have 2 versions, first illustrates the case where the excavated block is removed and second illustrates the case where the excavated block's properties are changed 

![image](https://user-images.githubusercontent.com/79562439/146336826-3c2b6a68-f855-4bda-bb9e-f7c44f7446a4.png)

The flow action is `PorousFlowUnsaturated`, and here is my `PorousFlowPiecewiseLinearSink` block (only allow for outflow)

```
   [./drain_face]
         type = PorousFlowPiecewiseLinearSinkTC
       variable = porepressure
       boundary = 'drain'
       pt_vals = '0 1e9' # x coordinates defining g
       multipliers = '0 1e9' # y coordinates defining g
       flux_function =  1e5 #1e5 # Variable C
       v = min
       use_mobility = true
       fluid_phase = 0
       use_relperm = true
    [../]
```
So below are the 3 test cases
- The first version (drain only at the right BC)
- The stitched model  (2nd case) with no change in permeability of the excavated block (drain the whole block)
- The stitched model (2nd case) with change in permeability of the excavated block to extremely high [*1e6] (drain the whole block)

below are the results, when I put all the phreatic surfaces together

![image](https://user-images.githubusercontent.com/79562439/146337907-038087fc-12bf-4c6b-a173-beb141f99631.png)

From top to bottom: no drain > The first version > The stitched model  (2nd case) with no change in permeability > The stitched model (2nd case) with change in permeability

![image](https://user-images.githubusercontent.com/79562439/146337787-fb5bbb80-c70a-4d43-b6dd-8100a9ae501a.png)

![image](https://user-images.githubusercontent.com/79562439/146337800-972dc892-0bf4-42cf-a079-5d83e4dd43d4.png)

![image](https://user-images.githubusercontent.com/79562439/146337809-d29057b7-658e-44f2-a697-73702876544e.png)

Theoretically, I thought the first case and the last case should produce the same results, did I miss anything? could you please explain what MOOSE does here so the results are different

Kind regards,
Traiwit

**Comment by Traiwit** [comment]
@GiudGiud do you know any expert on this topic? thanks!

**Reply by GiudGiud** [reply]
Only expert I know is tagged already on the post.

**Reply by Traiwit** [reply]
noted thanks!

**Comment by JerryLiu2023** [comment, marked as answer]
Traiwit:  I don't think you miss anything. Yes. When you set an extremely large permeability for the excavated block, theoretically, the first and third cases are the same. However, numerically, they are not. For those nodes on the right boundary for the first case, their nodal permeabilities (in permeability matrix in finite elements) are only contributed from left elements having lower permeabilities. Furthermore, these nodes are applied by prescribed flux.  In the third case, these nodes are now on the interface of two blocks having different permeabilities. Eventually, nodal permeabilies on these nodes are dominated by the right elements (extremely high perm). Furthermore, these nodes in the third case are interior nodes which are not applied by the prescribed flux. Consequently, pore pressure nearby (on the left side elements) numerically diffuses faster than the first case. If you use fine elements or higher-order elements, or hexa, their solutions are expected to be closer. Not only the permeability matrix but also the mass matrix in finite elements are different. I assume all your three cases are single fluid flow. If you have coupled solid mechanics field, the first and the third cases are totally different.  Hopefully, this helps.

**Reply by WilkAndy** [reply]
Thank you for answering @RLiu2020 .  I'm on holiday

**Reply by Traiwit** [reply]
Thank you @RLiu2020, yes these cases are single fluid flow with no mechanics coupling.

I'm wondering if there is a way to mimic the removal of the blocks without having to actually remove the blocks?
Removing the blocks disturbs our framework significantly as it renumbers the elements/nodes.

@WilkAndy I see that you changed the excav_perm to 0 in your mining example without having to remove the block, not sure if this is the case?

Kind regards,
Traiwit

**Reply by WilkAndy** [reply]
Sure, you can change the permeability in an element so no fluid flow happens there.  Whether that's a sensible thing to do depends on your boundary conditions!  The advantage to setting perm=0 is that no fluid will flow out of that block, which is good since it's not really there.

**Reply by Traiwit** [reply]
Hi @WilkAndy 

When I set-BC around the excavated block, I'd expect my pwp and the phreatic surface (water table) to be like this figure: 

![image](https://user-images.githubusercontent.com/79562439/148704702-f7175fc6-7ff3-486f-8878-a32f5999f0ae.png)

instead, the phreatic surface just wraps around the excavated block

```
     [./drain_DRAINSRF_CUMSRFSET_Y2012]
     type = PorousFlowPiecewiseLinearSinkTC
     variable = porepressure
     boundary = 'DRAINSRF_CUMSRFSET_Y2012'
     pt_vals = '0 1e9'
     multipliers = '0 1e9'
     flux_function = 1e8
     v = min
      use_mobility = true
      fluid_phase = 0
     [../]
```

Not sure if I've missed anything here

Kind regards,
Traiwit

**Reply by WilkAndy** [reply]
Your `drain_DRAINSRF_CUMSRFSET_Y2012` BC is draining if the porepressure is positive, but will remove zero water if the porepressure is negative.   Does this answer your question?

**Reply by Traiwit** [reply]
Hi @WilkAndy, yes , I was doing "only allow for outflow", as I thought water should not be re-entered to the geometry from the excavated blocks. Not sure if this is incorrect?

so here is from the real geometry, I was expecting the phreatic surface to follow the red line, but instead, it wraps tightly around the excavated blocks.

![image](https://user-images.githubusercontent.com/79562439/148705740-7d5063fa-991f-4bb9-853a-5c13bd5503c5.png)

**Reply by WilkAndy** [reply]
You are correct: water should not enter the unexcavated rock from the excavated rock.  However, what about if porepressure = -1E4 Pa in the rock?  Should it exit?  Usually, people answer:
(1) "yes, through capillarity and evaporation"
(2) "i don't care: i just care about the fully-saturated zone"
Perhaps this is a complete red-herring, but i just mention it because it may be important when attempting to accurately predict the water table.

Probably in your case, your insitu permeability is too low.  If you increase it, more water will flow to the excavation and lower the phreatic surface.

By the way, your `flux_function` is very high.  With `use_mobility = true`, i'd use something like `flux_function = 1`

a

**Reply by Traiwit** [reply]
@WilkAndy Thank you very much, I will try to work around things you suggested and get back to you.

Kind regards,
Traiwit

**Reply by WilkAndy** [reply]
Did you resolve this, @Traiwit ?

**Reply by Traiwit** [reply]
yep all good now, thanks!
